src/haddock/libs/libutil.py

```diff
@@ -293,6 +293,21 @@ def get_number_of_digits(num):
     return max(ceil(log10(num + 1)), 1)
 
 
+def get_zerofill_for_modules(modules):
+    """
+    Get a the prefix zerofill for modules.
+
+    If there are 5 modules, zerofill digits is 1.
+    If there are 10 modules, zerofill digits is 1 because counting
+    starts at 0 (for topoaa).
+    If there are 100 modules, zerofill digits is 2 because counting
+    starts at 0 (for topoaa).
+
+    This function is used in combination with `zero_fill`.
+    """
+    return max(1, get_number_of_digits(len(modules) - 1))
```

Comment:
Why not simply define everything with 3 digits? its already not realistic that 100 modules will be executed anyway

Reply:
would also make it easier to continue-restart. remember that the main guideline is sense and simplicity :P

Reply:
Yeah. That was the original idea. But @douweschulte yesterday rose a point on that, and he was right. There are drawbacks to having alternating leading zeros. But yesterday night I thought that it might not be that problem even for #383 after all. Anyway, now we need to accept this PR to finalize and correct #378. 

If in the future we decide to remove the alternating leading zeros, it is just a matter of changing a function name, nothing more.

Reply:
I would say it is nice to make sure that it can run in any at all times, even with stupidly high numbers of module, because this would be a very stupid reason to crash the execution of Haddock if everything else is just working. In this implementation if there are 100 modules it starts naming at `000` , `001`, .. . But even if you start a run with 2 modules and so get `0` and `1` it would still be very possible to just continue with `002` if you continue-restart with a run with many more modules.

Reply:
Yes. But some of these `000` numbers are hardcoded in the modules run files, for which if the name of the folders change, we would need to change also those paths. With `sed` most likely.

Reply:
But, I realized we need to address this path name changing for some examples, even if we use always two leading zeros. Therefore we can safely use the alternating zerofill; it should not hinder the other implementations.